File: 100.same-tree.js
/*
 * @lc app=leetcode id=100 lang=javascript
 *
 * [100] Same Tree
 */

function TreeNode(val, left, right) {
  this.val = val === undefined ? 0 : val;
  this.left = left === undefined ? null : left;
  this.right = right === undefined ? null : right;
}

// @lc code=start
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function (p, q) {
  if (!p && !q) {
    return true;
  }
  if ((p && !q) || (!p && q) || p.val !== q.val) {
    return false;
  }
  return isSameTree(p.left, q.left) && isSameTree(p.right, q.right);
};
// @lc code=end

File: 100.same-tree.go
/*
 * @lc app=leetcode id=100 lang=golang
 *
 * [100] Same Tree
 */

package leetcode

// @lc code=start
func isSameTree(p *TreeNode, q *TreeNode) bool {
	if p == nil && q == nil {
		return true
	}
	if (p != nil) != (q != nil) || p.Val != q.Val {
		return false
	}
	return isSameTree(p.Left, q.Left) && isSameTree(p.Right, q.Right)
}

// @lc code=end
